components/infobox/wikis/starcraft2/infobox_league_custom.lua

```diff
@@ -0,0 +1,602 @@
+		end
+		if premier > 0 then
+			playerBreakDown.display[#playerBreakDown.display + 1] = _PICON .. ' ' .. premier
+		end
+		if challenger > 0 then
+			playerBreakDown.display[#playerBreakDown.display + 1] = _CICON .. ' ' .. challenger
+		end
+	end
+	return playerBreakDown
+end
+
+function CustomLeague._playerRaceBreakDown(args)
+	local playerBreakDown = {}
+	local playerNumber = tonumber(args.player_number or 0) or 0
+	local zergNumber = tonumber(args.zerg_number or 0) or 0
+	local terranNumbner = tonumber(args.terran_number or 0) or 0
+	local protossNumber = tonumber(args.protoss_number or 0) or 0
+	local randomNumber = tonumber(args.random_number or 0) or 0
+	if playerNumber == 0 then
+		playerNumber = zergNumber + terranNumbner + protossNumber + randomNumber
+	end
+
+	if playerNumber > 0 then
+		playerBreakDown.playerNumber = playerNumber
+		if zergNumber + terranNumbner + protossNumber + randomNumber > 0 then
+			playerBreakDown.display = {}
+			if protossNumber > 0 then
+				table.insert(playerBreakDown.display, RaceIcon.getSmallIcon({'p'})
+					.. ' ' .. protossNumber)
+			end
+			if terranNumbner > 0 then
+				table.insert(playerBreakDown.display, RaceIcon.getSmallIcon({'t'})
+					.. ' ' .. terranNumbner)
+			end
+			if zergNumber > 0 then
+				table.insert(playerBreakDown.display, RaceIcon.getSmallIcon({'z'})
+					.. ' ' .. zergNumber)
+			end
+			if randomNumber > 0 then
+				table.insert(playerBreakDown.display, RaceIcon.getSmallIcon({'r'})
+					.. ' ' .. randomNumber)
+			end
+		end
+	end
+	Variables.varDefine('nbnotableP', protossNumber)
+	Variables.varDefine('nbnotableT', terranNumbner)
+	Variables.varDefine('nbnotableZ', zergNumber)
+	Variables.varDefine('nbnotableR', randomNumber)
+	return playerBreakDown
+end
+
+function CustomLeague._playerBreakDownDisplay(contents)
+	if type(contents) ~= 'table' or contents == {} then
+		return nil
+	end
+
+	local div = mw.html.create('div')
+	local number = #contents
+	for _, content in ipairs(contents) do
+		local infoboxCustomCell = mw.html.create('div'):addClass('infobox-cell-' .. number
+			.. ' infobox-center')
+		infoboxCustomCell:wikitext(content)
+		div:node(infoboxCustomCell)
+	end
+
+	return div
+end
+
+function CustomLeague:_makeBasedListFromArgs(args, base)
+	local firstArg = args[base .. '1']
+	local foundArgs = {CustomLeague:_makeInternalLink(firstArg)}
+	local index = 2
+
+	while not String.isEmpty(args[base .. index]) do
+		local currentArg = args['map' .. index]
+		table.insert(foundArgs, '&nbsp;• ' ..
+			tostring(CustomLeague:_createNoWrappingSpan(
+				CustomLeague:_makeInternalLink(currentArg)
+			))
+		)
+		index = index + 1
+	end
+
+	return foundArgs
+end
+
+function CustomLeague:defineCustomPageVariables(args)
+	--override vars that need custom handling on sc2
+	Variables.varDefine('tournament_game', string.lower(args.game or ''))
+	Variables.varDefine('tournament_series', mw.ext.TeamLiquidIntegration.resolve_redirect(args.series or ''))
```

Comment:
We should probably just make the resolving of redirect default

Reply:
would be okay for me (and sc2) but i think on some other wikis it might cause problems

Reply:
I think at the very least we should do it for LPDB, but I'm inclined to say vars though